apps/admin_api/lib/admin_api/v1/controllers/account_membership_controller.ex

```diff
@@ -28,13 +29,62 @@ defmodule AdminAPI.V1.AccountMembershipController do
   alias EWalletDB.{Account, Key, Membership, Role, User}
 
   @doc """
-  Lists the users that are assigned to the given account.
+  Lists the memberships for the given admin.
   """
-  def all_users_for_account(conn, attrs) do
+  def all_account_memberships_for_admin(conn, %{"id" => admin_id} = attrs) do
+    with %User{} = admin <-
+           User.get_admin(admin_id, preload: [memberships: [:user, :role]]) ||
+             {:error, :unauthorized},
+         {:ok, _} <- authorize(:get, conn.assigns, admin),
+         {:ok, %{query: query}} <- authorize(:all, conn.assigns, nil),
```

Comment:
The `query` returned can be `nil` if the actor has no permissions. Need to check with

```
true <- !is_nil(query) || {:error, :unauthorized} do
```

Can you add a test as well (with global role `none`)?